torch_geometric/nn/models/g_retriever.py

```diff
@@ -0,0 +1,236 @@
+        self.llm_dtype = llm_dtype
+        if llm_use_lora:
+            from peft import (
+                LoraConfig,
+                get_peft_model,
+                prepare_model_for_kbit_training,
+            )
+            self.llm_generator = prepare_model_for_kbit_training(
+                self.llm_generator)
+            lora_r: int = 8
+            lora_alpha: int = 16
+            lora_dropout: float = 0.05
+            lora_target_modules = [
+                "q_proj",
+                "v_proj",
+            ]
+            config = LoraConfig(
+                r=lora_r,
+                lora_alpha=lora_alpha,
+                target_modules=lora_target_modules,
+                lora_dropout=lora_dropout,
+                bias="none",
+                task_type="CAUSAL_LM",
+            )
+            self.llm_generator = get_peft_model(self.llm_generator, config)
+        self.llm_device = self.llm_to_use.llm_device
+        self.tokenizer = self.llm_to_use.tokenizer
+        try:
+            self.graph_encoder = gnn_to_use(
+                in_channels=gnn_in_channels,
+                out_channels=gnn_out_channels,
+                hidden_channels=gnn_hidden_channels,
+                num_layers=num_gnn_layers,
+                heads=num_gnn_heads,
+                norm='batch_norm',
+            ).to(self.llm_device)
+        except:  # noqa: E722
+            # to handle gnns that do not have `heads` param
+            self.graph_encoder = gnn_to_use(
+                in_channels=gnn_in_channels,
+                out_channels=gnn_out_channels,
+                hidden_channels=gnn_hidden_channels,
+                num_layers=num_gnn_layers,
+                norm='batch_norm',
+            ).to(self.llm_device)
+        # For the MLP Projection
+        mlp_hidden_dim = gnn_out_channels
+        self.projector = nn.Sequential(
+            nn.Linear(gnn_out_channels, mlp_hidden_dim),
+            nn.Sigmoid(),
+            nn.Linear(mlp_hidden_dim, mlp_out_dim),
+        ).to(self.llm_device)
+
+        self.word_embedding = self.llm_to_use.word_embedding
+
+    def encode_graphs(self, node_feat, edge_index, edge_attr, batch):
+        x = node_feat.to(self.llm_device)
+        edge_index = edge_index.long().to(self.llm_device)
+        edge_attr = edge_attr.to(self.llm_device)
+        n_embeds = self.graph_encoder(x, edge_index.long(), edge_attr)
+        batch = batch.to(self.llm_device)
+        g_embeds = scatter(n_embeds, batch, dim=0, reduce='mean')
+        return g_embeds
+
+    def forward(
+        self,
+        question: List[str],
+        node_feat: torch.Tensor,
+        edge_index: torch.Tensor,
+        batch: torch.Tensor,
+        ptr: torch.Tensor,
+        label: List[str],
+        edge_attr: Optional[torch.Tensor] = None,
+        additional_text_context: Optional[List[str]] = None,
+    ):
+        r"""Forward pass.
+
+        Args:
+            question (List[str]): The questions/prompts.
```

Comment:
Does this means that passing a list of questions / prompts and they will use the same subgraph?

Reply:
no each q/prompt is matched to a subgraph, they are just combined by the pyg dataloader collate step